File: 2107-find-unique-binary-string/find-unique-binary-string.js
var findDifferentBinaryString = function(nums) {
    let n = nums[0].length;
    let numSet = new Set(nums);

    function generateBinary(binary) {
        if (binary.length === n) {
            if (!numSet.has(binary)) {
                return binary;
            }
            return null;
        }
        let res = generateBinary(binary + '0');
        if (res) return res;
        return generateBinary(binary + '1');
    }

    return generateBinary("");
};

File: 2107-find-unique-binary-string/find-unique-binary-string.dart
class Solution {
  String findDifferentBinaryString(List<String> nums) {
    int n = nums[0].length;
    String ans = "";

    void generateBinary(String binary) {
      if (binary.length == n) {
        if (!nums.contains(binary)) {
          ans = binary;
        }
        return;
      }
      if (ans.isNotEmpty) return;
      generateBinary(binary + '0');
      generateBinary(binary + '1');
    }

    generateBinary('');
    return ans;
  }
}